include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
     D3D12_NODE_MASK m_Desc;
 };
 
+//------------------------------------------------------------------------------------------------
+// Peter: Implementation for the new CheckFeatureSupport API
+
+#define FEATURE_SUPPORT_GET(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION() const \
+{ \
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NAME(RETTYPE,FEATURE,OPTION,NAME) \
+RETTYPE NAME() const \
+{\
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NODE_INDEXED(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION(UINT NodeIndex=0) const \
+{\
+    return FEATURE[NodeIndex].OPTION; \
+}
+
+#define INITIALIZE_MEMBER_DATA_CHECKED(FEATURE, MEMBER) \
+if (FAILED(result = m_pDevice->CheckFeatureSupport(FEATURE, &MEMBER, sizeof(MEMBER)))) { \
+    return result; \
+}
+
+class CD3DX12FeatureSupport
+{
+public:
+    CD3DX12FeatureSupport(ID3D12Device* pDevice)
+    : m_pDevice(pDevice)
+    {
+        m_hStatus = Init();
+    }
+
+    HRESULT Init()
+    {
+        // TODO: Think about hot to catch and report errors
+        HRESULT result;
+        
+        // Initialize static feature support data structures
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_D3D12_OPTIONS, m_dOptions);
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_GPU_VIRTUAL_ADDRESS_SUPPORT, m_dGPUVASupport);
+        m_dShaderModel.HighestShaderModel = D3D_HIGHEST_SHADER_MODEL; // TODO: What to do if highest model is inconsistent in the DLL and in the header?
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_SHADER_MODEL, m_dShaderModel);
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_D3D12_OPTIONS1, m_dOptions1);
+        m_dRootSignature.HighestVersion = D3D_HIGHEST_ROOT_SIGNATURE_VERSION;
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_ROOT_SIGNATURE, m_dRootSignature);
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_D3D12_OPTIONS2, m_dOptions2);
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_SHADER_CACHE, m_dShaderCache);
+
+        // Initialize per-node feature support data structures
+        const UINT uNodeCount = m_pDevice->GetNodeCount();
+        m_dProtectedResourceSessionSupport.resize(uNodeCount);
+        m_dArchitecture1.resize(uNodeCount);
+        for (UINT i = 0; i < uNodeCount; i++) {
+            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_PROTECTED_RESOURCE_SESSION_SUPPORT, m_dProtectedResourceSessionSupport[i]);
+            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i]);
+        }
+
+        return result;
+    }
+
+    HRESULT GetStatus() const
+    {
+        return m_hStatus;
+    }
+
+    static std::variant<CD3DX12FeatureSupport, HRESULT> Create(ID3D12Device* pDevice)

```

Comment:
This is the factory method as discussed in the team chat. It returns either the initialized FeatureSupport intance on success or the HRESULT error code on failure.